modern/src/runtime/MakiObject.ts

```diff
@@ -14,6 +14,7 @@ class MakiObject {
   _emitter: Emitter;
   children: MakiObject[];
   js_annotations: Object;
+  system: MakiObject | undefined;
```

Comment:
I'm not sure how we should type this. It should be `System`, but `System` is a subclass of `MakiObject`. Maybe there is a different/better way to keep track of the `System` objects we generate to unload scripts at the end, thoughts?

Reply:
I guess we could just create an array of `System`s and toss it in redux, but I'm not sure

Reply:
We don't need this to be present on every maki object right? Just Script objects I think. And once we get to a [Mutable XML Tree](https://paper.dropbox.com/doc/A-Third-Way-Mutable-XML-Tree-vx3iPfGIBmSHEDJSMh0bn) world, it can just live on the XML node itself rather than the fake maki object.

Reply:
I think for now the simplest thing might be to create a `JSScript` or whatever object that inherits from MakiObject and then only it will have the prop and we can properly type it